core/src/KokkosExp_MDRangePolicy.hpp

```diff
@@ -62,6 +64,46 @@
 #include <ROCm/KokkosExp_ROCm_IterateTile_Refactor.hpp>
 #endif
 
+namespace Kokkos {
+namespace Impl {
+
+// Workaround for narrowing conversion warnings
+template <class IndexType>
+struct _ignore_narrowing_index_wrapper {
+  IndexType value;
+
+  template <class T,
+            int = typename std::enable_if<
+                std::is_convertible<T, IndexType>::value, int>::type{0}>
+  /* intentionally implicit */
+  KOKKOS_FORCEINLINE_FUNCTION _ignore_narrowing_index_wrapper(
+      T arg_value) noexcept
+      : value(arg_value) {}
+
+  // workaround for Cuda linker bug??!?
```

Comment:
Is there a bug or is there not?
Would you remind me why the default constructor cannot be declared as defaulted?
And why do we absolutely have to write all the boiler plate code below?
Can't you just get away with that constructor and the implicit conversions to the underlying index type.

Reply:
It can be defaulted, but there was some bug in `nvcc` that was causing the linker not to find the symbol (even if it was force-inlined).  This and another thing below seemed to fix the issue